dockerfiles/QAT/run.sh

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+set -u
+
+# NOTE: Some env variables are set in the Dockerfile - those that are
+# intended to be over-rideable.
+QAT_SRC=~/src/QAT
+export GOPATH=~/src/go
+export PATH=${PATH}:/usr/local/go/bin:${GOPATH}/bin
+
+packagerepo=github.com/kata-containers/packaging
+packagerepopath=${GOPATH}/src/${packagerepo}
+
+osbrepo=github.com/kata-containers/osbuilder
+osbrepopath=${GOPATH}/src/${osbrepo}
+
+testsrepo=github.com/kata-containers/tests
+testsrepopath=${GOPATH}/src/${testsrepo}
+
+runtimerepo=github.com/kata-containers/runtime
+runtimerepopath=${GOPATH}/src/${runtimerepo}
+
+agentrepo=github.com/kata-containers/agent
+agentrepopath=${GOPATH}/src/${agentrepo}
+
+grab_kata_repos()
+{
+	# Check out all the repos we will use now, so we can try and ensure they use the specified branch
+	# Only check out the branch needed, and make it shallow and thus space/bandwidth efficient
+	git clone --single-branch --branch $KATA_VERSION --depth=1 https://${packagerepo} ${packagerepopath}
```

Comment:
This failed for me because it did not pick up my proxy information.

Reply:
Can you show the exact error you got? I want to check if that was git'ing via https or the git: protocol. It should afaik be using the https one, in which case it should pick up afaik your docker (host) configured proxy info.

Reply:
Both my $http_proxy and $https_proxy variables are set. 

```
$ docker run -ti --rm --privileged -v /dev:/dev -v $(pwd)/output:/output kataqat
Cloning into '/root/src/go/src/github.com/kata-containers/packaging'...
fatal: unable to access 'https://github.com/kata-containers/packaging/': Failed to connect to github.com port 443: Connection timed out
```

If I git clone on the host it works without issue.  In my next comment I suggested to clone as part of the Dockerfile. That might help with this issue as well. You are doing the clone from a script inside a docker container vs on the Docker ADD line as part of building the container. Someone can successfully build a Docker image but then that image won't connect to anything because of proxies issues and it will be harder to debug. However, if the clones are done as part of the ADD command and they fail there then the container image won't get built and it is more obvious it is a proxy connection issue that can be fixed by passing the http_proxy and https_proxy variables on the `docker build` command.

Reply:
> Both my $http_proxy and $https_proxy variables are set.

Set where - in your host env, or in your docker config?

Reply:
It is set in both.

```
$ echo $http_proxy
http://proxy:port/
$ echo $https_proxy
http://proxy:port//


$ cat /etc/systemd/system/docker.service.d/proxy.conf
[Service]
Environment="HTTP_PROXY=http://proxy:port/"
Environment="HTTPS_PROXY=http://proxy:port/"
```

Reply:
Ah, the bit you are maybe missing is to get the proxy settings *inside* the containers. The above I believe sets them:
- on the host
- in the systemd unit of the docker server

I then placed this in my `~/.docker/config.json`:

```json
{
 "proxies":
 {
   "default":
   {
     "httpProxy": "http://proxy:port",
     "httpsProxy": "http://proxy:port"
   }
 }
}
```

which gets the proxy information *into* the containers themselves, and then the image ran and built for me. You want to try that @eadamsintel ?

Reply:
> I then placed this in my `~/.docker/config.json`

This is the only sane setting when working with proxies.

Reply:
I put the proxies in both locations and can build and run containers successfully.